### PR TITLE
Update devcontainer schema

### PR DESCRIPTION
**Pull request by joshaber**
Update the devcontainer schema to support parallel lifecycle scripts.

See https://github.com/devcontainers/spec/blob/main/proposals/parallel-lifecycle-script-execution.md for more information.

### CODE REVIEW

**Comment by joshaber** [comment]
cc @chrmarti

File: extensions/configuration-editing/schemas/devContainer.schema.src.json [hunk 1]
```diff
@@ -253,51 +253,101 @@
 				"onCreateCommand": {
 					"type": [
 						"string",
-						"array"
+						"array",
+						"object"
 					],
 					"description": "A command to run when creating the container. This command is run after \"initializeCommand\" and before \"updateContentCommand\". If this is a single string, it will be run in a shell. If this is an array of strings, it will be run as a single command without shell.",
```
Comment (chrmarti) [inline extensions/configuration-editing/schemas/devContainer.schema.src.json:259]: Is there a short way to describe the additional option in the description? Maybe "Alternatively an object with the property keys as labels and the values as either strings or arrays of strings can be used to configure multiple commands."? (We reuse the wording of the description for the other commands.)

**Review by chrmarti — CHANGES_REQUESTED** [review]

File: extensions/configuration-editing/schemas/devContainer.schema.src.json (hunk 1 above)
Comment (joshaber) [reply extensions/configuration-editing/schemas/devContainer.schema.src.json]: Good call, updated 👍

File: extensions/configuration-editing/schemas/devContainer.schema.src.json [hunk 2]
```diff
@@ -253,51 +253,101 @@
 				"onCreateCommand": {
 					"type": [
 						"string",
-						"array"
+						"array",
```
[the remark this answers is not in the document]
Comment (joshaber) [reply extensions/configuration-editing/schemas/devContainer.schema.src.json:256]: I don't recall specifically excluding it, I think it's a gap in my implementation 🙈

[the remark this answers is not in the document]
Comment (chrmarti) [reply extensions/configuration-editing/schemas/devContainer.schema.src.json:256]: Should we still do that? (In separate PRs, I'd suggest.)

File: extensions/configuration-editing/schemas/devContainer.schema.src.json [hunk 3]
```diff
@@ -253,51 +253,101 @@
...
+							"string",
+							"array"
+						],
+						"items": {
+							"type": "string"
+						}
 					}
 				},
 				"updateContentCommand": {
 					"type": [
```
Comment (chrmarti) [inline extensions/configuration-editing/schemas/devContainer.schema.src.json:274]: @aeschli You recommended using "oneOf" in a similar case. Does this apply here too? Would "string" and "array" go in one clause and "object" in another? What is the advantage of "oneOf"? Thanks!

**Review by chrmarti — CHANGES_REQUESTED** [review]

File: extensions/configuration-editing/schemas/devContainer.schema.src.json (hunk 3 above)
Comment (chrmarti) [reply extensions/configuration-editing/schemas/devContainer.schema.src.json:274]: Chatted with @aeschli and we can include this as-is. oneOf is easier to support on the tooling side, but I can revisit when we find this is important. Thanks.

**Review by chrmarti — APPROVED** [review]

**Pull request merged**